lib/util/cachedMerge.js

```diff
@@ -6,19 +6,14 @@
 
 const mergeCache = new WeakMap();
 
-const cachedMerge = (first, ...args) => {
-	if (args.length === 0) return first;
-	if (args.length > 1) {
-		return cachedMerge(first, cachedMerge(...args));
```

Comment:
It was so cool...

Reply:
But not correct!

https://github.com/Microsoft/TypeScript/issues/22975#issuecomment-380546619